pkg/genericapiserver/storage_factory.go

```diff
@@ -219,3 +226,25 @@ func (s *DefaultStorageFactory) Backends() []string {
 	}
 	return backends.List()
 }
+
+// NewStorageCodec assembles a storage codec for the provided storage media type, the provided serializer, and the requested
+// storage and memory versions.
+func NewStorageCodec(storageMediaType string, ns runtime.StorageSerializer, storageVersion, memoryVersion unversioned.GroupVersion) (runtime.Codec, error) {
+	s, ok := ns.SerializerForMediaType(storageMediaType, nil)
+	if !ok {
+		return nil, fmt.Errorf("unable to find serializer for JSON")
```

Comment:
Fixed in a later commit (spotted that as well)

On Mon, Apr 25, 2016 at 11:10 AM, David Eads <notifications@github.com>
wrote:

> In pkg/genericapiserver/storage_factory.go
> <https://github.com/kubernetes/kubernetes/pull/24710#discussion_r60929741>
> :
>
> > @@ -219,3 +226,25 @@ func (s *DefaultStorageFactory) Backends() []string {
> >  	}
> >  	return backends.List()
> >  }
> > +
> > +// NewStorageCodec assembles a storage codec for the provided storage media type, the provided serializer, and the requested
> > +// storage and memory versions.
> > +func NewStorageCodec(storageMediaType string, ns runtime.StorageSerializer, storageVersion, memoryVersion unversioned.GroupVersion) (runtime.Codec, error) {
> > +	s, ok := ns.SerializerForMediaType(storageMediaType, nil)
> > +	if !ok {
> > +		return nil, fmt.Errorf("unable to find serializer for JSON")
>
> not necessarily json
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/24710/files/63ae8755df5ae6beaaab9dd386d019b619165e40..c612ff15f95d5ec56d4a57b169b7066823c146ba#r60929741>
>